control/controlhttp/client.go

```diff
@@ -53,6 +61,9 @@ const (
 //
 // If Dial fails to connect using addr, it also tries to tunnel over
 // TLS to <addr's host>:443 as a compatibility fallback.
+//
+// The provided ctx is only used for the initial connection, until
+// Dial returns. It does not affect the connection once established.
```

Comment:
Could you throw a TODO in here to do something with conn read/write timeouts? I think I neglected to plumb the ctx deadline all the way into the socket, so there are some ways this package (and possibly also controlbase?) can be slowlorised at the moment.